dlc/src/main/scala/org/bitcoins/dlc/DLCClient.scala

```diff
@@ -238,6 +239,39 @@ case class DLCClient(
     sigsMapF.map(FundingSignatures.apply)
   }
 
+  def verifyRemoteFundingSigs(remoteSigs: FundingSignatures): Boolean = {
+    val remoteTweak = if (isInitiator) {
+      fundingUtxos.length
+    } else {
+      0
+    }
+
+    val psbt = PSBT.fromUnsignedTx(createUnsignedFundingTransaction)
+
+    remoteSigs.zipWithIndex
+      .foldLeft(true) {
+        case (ret, ((outPoint, sigs), index)) =>
+          require(psbt.transaction.inputs(index).previousOutput == outPoint,
+                  "Adding signature for incorrect input")
+
+          val idx = index + remoteTweak
+
+          psbt
+            .addWitnessUTXOToInput(remoteFundingInputs(index).output, idx)
+            .addSignatures(sigs, idx)
+            .inputMaps(idx)
+            .finalize(psbt.transaction.inputs(idx)) match {
+            case Success(inputMap) =>
+              val newInputMaps = psbt.inputMaps.updated(idx, inputMap)
+              val finalized = psbt.copy(inputMaps = newInputMaps)
+
+              ret && finalized.verifyFinalizedInput(idx)
```

Comment:
should probably make this whole thing an `if` statement on `ret` rather than putting it at the bottom so that you don't do a ton of work if the first one fails

Reply:
This shouldn't be needed because scala has short circuiting implemented, so if the first part of the `&&` is false, it won't evaluate the rest. Also with `||`, if the first part is `true` then it won't evaluate the rest.

Reply:
From the docs
![image](https://user-images.githubusercontent.com/15256660/79494347-79c03000-7fe8-11ea-8bcd-040611565e29.png)

Reply:
actually seems `foldLeft` does not use short-circuiting however, still applies to other comments

Reply:
Scala does have short circuiting but that only applies to the second argument to `&&`, not to all of the code that constructs `finalized` and `idx` here (analogous statement elsewhere). So unless you want to make all the `val`s `lazy`, which I think would be a little weird to read, I'd suggest using `if`

Reply:
Yes, I moved this to an if because of that